src/components/dialogs/parameters/short-circuit-parameters.js

```diff
@@ -56,181 +66,179 @@ export const useGetShortCircuitParameters = () => {
     return [shortCircuitParams, setShortCircuitParams];
 };
 
-function getValue(param, key) {
-    if (!param || param[key] === undefined) {
-        return null;
+const formSchema = yup
+    .object()
+    .shape({
+        [SHORT_CIRCUIT_WITH_FEEDER_RESULT]: yup.boolean(),
+        [SHORT_CIRCUIT_PREDEFINED_PARAMS]: yup.string().required(),
+        [SHORT_CIRCUIT_WITH_LOADS]: yup.boolean(),
+        [SHORT_CIRCUIT_WITH_VSC_CONVERTER_STATIONS]: yup.boolean(),
+        [SHORT_CIRCUIT_WITH_SHUNT_COMPENSATORS]: yup.boolean(),
+        [SHORT_CIRCUIT_WITH_NEUTRAL_POSITION]: yup.boolean(),
+        [SHORT_CIRCUIT_INITIAL_VOLTAGE_PROFILE_MODE]: yup.string().required(),
+    })
+    .required();
+
+const prepareDataToSend = (shortCircuitParams, newParameters) => {
+    const oldParameters = { ...shortCircuitParams.parameters };
+    let parameters = { ...oldParameters, ...newParameters };
+    // we need to add voltageRanges to the parameters only iwhen INITIAL_TENSION is configured
+    if (
+        newParameters.initialVoltageProfileMode === INITIAL_TENSION.CONFIGURED
+    ) {
+        parameters = {
+            ...parameters,
+            voltageRanges: shortCircuitParams.voltageRangesInfo.CONFIGURED,
+        };
     }
-    return param[key];
-}
-
-const DropDown = ({ value, label, values, callback }) => {
-    return (
-        <>
-            <Grid item xs={8} sx={styles.parameterName}>
-                <FormattedMessage id={label} />
-            </Grid>
-            <Grid item container xs={4} sx={styles.controlItem}>
-                <Select
-                    labelId={label}
-                    value={value}
-                    onChange={callback}
-                    size="small"
-                >
-                    {Object.keys(values).map((key) => (
-                        <MenuItem key={key} value={key}>
-                            <FormattedMessage id={values[key]} />
-                        </MenuItem>
-                    ))}
-                </Select>
-            </Grid>
-        </>
-    );
-};
+    // we need to remove the predefinedParameters attribut from parameters because it's not handled on powsybl class
```

Comment:
Why remove ? You send ShortCircuitParametersInfos, not ShortCircuitParameters

Reply:
When the form is submitted the newParameters contains all the fields. Amongst  these fields "predefinedParameters" which is not supported by ShortCircuitParameters.
So first I create **parameters** that contains all needed fields for  ShortCircuitParameters and then I remove the  predefinedParameters from **parameters** because as I said it is not a field of ShortCircuitParameters.
I could remove predefinedParameters from the beginning but it's the same thing.